core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts

```diff
@@ -228,6 +250,38 @@ export class DotEmaComponent implements OnInit, OnDestroy {
 
                 this.savePayload = payload;
             },
+            [CUSTOMER_ACTIONS.DELETE_CONTENTLET]: () => {
+                const { pageContainers, container, contentletId, pageID } = <
+                    DeleteContentletPayload
+                >data.payload;
+
+                const newPageContainers = deleteContentletFromContainer({
+                    pageContainers: pageContainers,
+                    container: container,
+                    contentletID: contentletId
+                });
+
+                this.confirmationService.confirm({
+                    header: this.dotMessageService.get(
+                        'editpage.content.contentlet.remove.confirmation_message.header'
+                    ),
+                    message: this.dotMessageService.get(
+                        'editpage.content.contentlet.remove.confirmation_message.message'
+                    ),
+                    acceptLabel: this.dotMessageService.get('dot.common.dialog.accept'),
+                    rejectLabel: this.dotMessageService.get('dot.common.dialog.reject'),
+                    accept: () => {
+                        this.store.savePage({
+                            pageContainers: newPageContainers,
+                            pageID,
+                            whenSaved: () => {
+                                this.resetDialogIframeData();
+                                this.reloadIframe();
+                            }
+                        }); // Save when selected
```

Comment:
Should we clean the `savePayload` in the cancel or close?

Reply:
In the delete we don't make use of the `savePayload` property